### DISCUSSION TITLE
API using unix socket

### DISCUSSION THREAD

**Discussion by sergejostir** [drakkan/sftpgo#266] [Ideas]
Would it be possible to add an option to expose unix socket instead of binding httpd to a port? I will only access API from local services and it's kinda a chore to occupy additional ports for that.

Other services like MySQL, Docker daemon, etc all support being accessed using unix sockets instead of TCP connections.

**Comment by drakkan** [comment]
This is a good suggestion thank you. I'll give a try

**Reply by drakkan** [reply]
> Would it be possible to add an option to expose unix socket instead of binding httpd to a port? I will only access API from local services and it's kinda a chore to occupy additional ports for that.
> 
> Other services like MySQL, Docker daemon, etc all support being accessed using unix sockets instead of TCP connections.

This should work [now](https://github.com/drakkan/sftpgo/commit/0966d44c0f350e0ea23bcc1d2542dd6c5f21eb8d), you case set a `bind_address` such as this one:

```
"bind_address": "/tmp/sftpgo.sock"
```

I tested both http and https using curl as client: `curl -k -v --unix-socket /tmp/sftpgo.sock http://sock/healthz`
